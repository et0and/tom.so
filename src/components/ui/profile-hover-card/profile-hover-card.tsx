import { CalendarDays } from "lucide-react";
import {
  Avatar,
  AvatarFallback,
  AvatarImage,
} from "@/components/ui/avatar/avatar";
import {
  HoverCard,
  HoverCardContent,
  HoverCardTrigger,
} from "@/components/ui/card/hover-card";

interface ProfileHoverCardProps {
  readonly href: string;
  readonly linkTitle: string;
  readonly avatarSrc: string;
  readonly avatarFallback: string;
  readonly title: string;
  readonly description: string;
  readonly joinedDate: string;
  readonly className?: string;
}

export function ProfileHoverCard({
  href,
  linkTitle,
  avatarSrc,
  avatarFallback,
  title,
  description,
  joinedDate,
  className = "link",
}: ProfileHoverCardProps) {
  return (
    <HoverCard>
      <HoverCardTrigger asChild>
        <a href={href} className={className}>
          {linkTitle}
        </a>
      </HoverCardTrigger>
      <HoverCardContent className="w-80">
        <div className="flex justify-between space-x-4">
          <Avatar>
            <AvatarImage src={avatarSrc} />
            <AvatarFallback>{avatarFallback}</AvatarFallback>
          </Avatar>
          <div className="space-y-1">
            <p className="text-sm font-semibold">{title}</p>
            <p className="text-sm">{description}</p>
            <div className="flex items-center pt-2">
              <CalendarDays className="mr-2 h-4 w-4 opacity-70" />{" "}
              <span className="text-sm text-muted-foreground">
                {joinedDate}
              </span>
            </div>
          </div>
        </div>
      </HoverCardContent>
    </HoverCard>
  );
}
