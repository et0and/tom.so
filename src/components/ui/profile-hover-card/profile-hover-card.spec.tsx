import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { ProfileHoverCard } from "./profile-hover-card";

describe("<ProfileHoverCard />", () => {
  it("should render correctly", () => {
    const { container } = render(
      <ProfileHoverCard
        href="https://example.com"
        linkTitle="Example Link"
        avatarSrc="/avatar.jpg"
        avatarFallback="EX"
        title="Example Title"
        description="Example description text"
        joinedDate="Joined December 2023"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with custom className", () => {
    const { container } = render(
      <ProfileHoverCard
        href="https://example.com"
        linkTitle="Example Link"
        avatarSrc="/avatar.jpg"
        avatarFallback="EX"
        title="Example Title"
        description="Example description text"
        joinedDate="Joined December 2023"
        className="custom-link"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
