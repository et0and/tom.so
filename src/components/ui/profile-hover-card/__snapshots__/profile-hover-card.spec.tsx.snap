// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProfileHoverCard />S > should render correctly 1`] = `
<div>
  <a
    class="link"
    data-state="closed"
    href="https://example.com"
  >
    Example Link
  </a>
</div>
`;

exports[`<ProfileHoverCard />S > should render with custom className 1`] = `
<div>
  <a
    class="custom-link"
    data-state="closed"
    href="https://example.com"
  >
    Example Link
  </a>
</div>
`;
